editor.planx.uk/src/pages/FlowEditor/components/Sidebar/LinkDialog.tsx

```diff
@@ -0,0 +1,302 @@
+  type: "published" | "draft" | "preview";
+  isPublished?: boolean;
+}
+
+type PublishedLinkProps = Pick<
+  LinkProps,
+  "status" | "subdomain" | "link" | "isPublished"
+>;
+
+const InactiveLink = styled(Typography)(({ theme }) => ({
+  width: "100%",
+  textAlign: "left",
+  color: theme.palette.text.secondary,
+}));
+
+const PaddedText = styled(Typography)(({ theme }) => ({
+  paddingLeft: "31px",
+}));
+
+const LinkBox = styled(Box)(({ theme }) => ({
+  display: "flex",
+  flexDirection: "column",
+  gap: "10px",
+}));
+
+const CopyButton = (props: any) => {
+  const [copyMessage, setCopyMessage] = useState<"copy" | "copied">("copy");
+  return (
+    <Tooltip title={copyMessage}>
+      <Button
+        component={"button"}
+        variant="help"
+        onMouseLeave={() => {
+          setTimeout(() => {
+            setCopyMessage("copy");
+          }, 500);
+        }}
+        onClick={() => {
+          setCopyMessage("copied");
+          navigator.clipboard.writeText(props.link);
+        }}
+        sx={{ marginLeft: "8px" }}
+      >
+        <Typography
+          display={"flex"}
+          flexDirection={"row"}
+          gap={"4px"}
+          variant="body3"
+        >
+          <ContentCopyIcon style={{ width: "18px", height: "18px" }} />
+          {copyMessage}
+        </Typography>
+      </Button>
+    </Tooltip>
+  );
+};
+
+const InactivePublishedLink = (props: any) => {
+  return (
+    <LinkBox>
+      {" "}
+      <PaddedText variant="h4">{props.title}</PaddedText>
+      <InactiveLink pl={"31px"}>{props.message}</InactiveLink>{" "}
+    </LinkBox>
+  );
+};
+
+const ActivePublishedLink = (props: any) => {
+  return (
+    <LinkBox>
+      {" "}
+      <PaddedText variant="h4">
+        {props.title} <CopyButton link={props.link} />
+      </PaddedText>
+      <Link pl="31px" href={props.link}>
+        {props.link}
+      </Link>
+    </LinkBox>
+  );
+};
+
+const PublishedLinkContent = (props: PublishedLinkProps) => {
+  return (
+    <Stack spacing={"15px"}>
+      {props.subdomain && props.status === "online" && props.isPublished ? (
+        <ActivePublishedLink title="With subdomain" link={props.subdomain} />
+      ) : props.subdomain && props.status === "online" && !props.isPublished ? (
+        <InactivePublishedLink
+          title="With subdomain"
+          message={props.subdomain}
+        />
+      ) : props.subdomain && props.status === "offline" && props.isPublished ? (
+        <InactivePublishedLink
+          title="With subdomain"
+          message={props.subdomain}
+        />
+      ) : (
+        <InactivePublishedLink
+          title="With subdomain"
+          message={"There is not a domain configured for this team"}
+        />
+      )}
+      {props.isPublished && props.status === "online" ? (
+        <ActivePublishedLink title="PlanX link" link={props.link} />
+      ) : (
+        <InactivePublishedLink title="PlanX link" message={props.link} />
+      )}
```

Comment:
I find this ternary very difficult to read, and similar to another comment would much prefer more verbose `if/else` or `switch` syntax here to describe the expected conditional rendering rules. 

Also a great example where tests would help describe expected behavior here !